app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -125,6 +128,16 @@ open class HomeActivity : AppCompatActivity() {
         }
     }
 
+    final override fun onRestart() {
+        hotStartMonitor.onRestartFirstMethodCall()
+        super.onRestart()
+    }
+
+    final override fun onPostResume() {
+        super.onPostResume()
+        hotStartMonitor.onPostResumeFinalMethodCall()
+    }
+
```

Comment:
I am convinced that this captures what we want. That said, I *hate* the lifecycle and reserve the right to later say, "Yes, of course," when we discover that this somehow misses a case.

Reply:
That's fair – I totally recognize measurement is making a best effort and iterating as necessary (e.g. to reduce noise) and Android doesn't make that any easier. :)